internal/server/controller/controller.go

```diff
@@ -1,15 +1,19 @@
 package controller
 
 import (
+	"context"
 	"fmt"
 	"github.com/go-chi/chi/v5"
 	sapp "github.com/maybecoding/go-metrics.git/internal/server/app"
+	compress2 "github.com/maybecoding/go-metrics.git/pkg/compress"
```

Comment:
линтер обычно ругается на алиасы, да и смысла в них нет, лучше переменные придумтаь чтобы не пересекались
```suggestion
	"github.com/maybecoding/go-metrics.git/pkg/compress"
```